src/main/kotlin/dev/arbjerg/ukulele/audio/Player.kt

```diff
@@ -101,7 +103,15 @@ class Player(val beans: Beans, guildProperties: GuildProperties) : AudioEventAda
         player.stopTrack()
     }
 
+    fun toggleRepeat() {
+        repeatOne = !repeatOne
+    }
+
     override fun onTrackEnd(player: AudioPlayer, track: AudioTrack, endReason: AudioTrackEndReason) {
+        if (repeatOne) {
+            queue.addNext(track.makeClone())
```

Comment:
This is where the actual repeat one takes place.  So when a track is ending, if repeatOne is set, it will add a clone of that track that's ending to the queue as the next track (zeroth index).